src/providers/dataProvider/index.ts

```diff
@@ -62,6 +62,23 @@ export const getDataProvider = async (): Promise<DataProvider<string>> => {
   )
   const providerWithCustomMethods = { ...provider }
   const audit = trackEvent(providerWithCustomMethods)
+
```

Comment:
1. Could you provide a return type for the method please?
2. Why not filter the records in the initial `getList`?